assets/customElements/map_filter_form.js

```diff
@@ -0,0 +1,35 @@
+// @ts-check
+
+
+// credits to https://github.com/fairnesscoop/permacoop/blob/master/src/assets/customElements/autoForm.js
+export default class extends HTMLElement {
```

Comment:
Je ne comprends pas trop l'usage de cette fonction, pourrais-tu m'expliquer stp ?

Reply:
Je confirme l'intention par contre l'implémentation me paraît bizarrement compliquée

Qu'est-ce qui empêche de récupérer une référence au `<form>` puis de faire `for (const formControl of form.elements)` comme dans l'autoForm de Permacoop ?

Pour la modification de l'URL, les navigateurs ont un comportement par défaut qui consiste à mettre les champs idenifiés par leur `name` dans l'URL car on navigue via un `<form method="GET">`. Vu qu'on est sur une page dynamique où on veut garder la carte on ne peut pas l'utiliser directement, mais j'intuitionne que Turbo permet de s'appuyer là dessus. Avec `data-turbo-action="advance"` je pense : https://turbo.hotwired.dev/handbook/frames#promoting-a-frame-navigation-to-a-page-visit

Reply:
Indeed, le `data-turbo-action="advance"` permet de simplifier grandement les choses. Ca fonctionne pour les liens classiques, j'imagine que ce sera aussi le cas pour les forms.